### PR TITLE
[W7][T09-1] Yeo Cheng Hong

### PR DESCRIPTION
**Pull request by chyeo**
A step by step implementation to allow the `sort` command to be accepted by the program.

The `sort` commands sorts the current users in the address book in alphabetical order.

The user guide and test cases have been updated accordingly to reflect the `sort` command updates.

### CODE REVIEW

File: src/seedu/addressbook/commands/SortCommand.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package seedu.addressbook.commands;
+
+public class SortCommand {
```
Comment (Creastery) [inline src/seedu/addressbook/commands/SortCommand.java]: Perhaps you could refine the commit subject title and message body further to indicate that this is an unused file!

File: src/seedu/addressbook/commands/SortCommand.java [hunk 2]
```diff
@@ -1,4 +1,26 @@
 package seedu.addressbook.commands;
 
-public class SortCommand {
+import seedu.addressbook.data.person.ReadOnlyPerson;
+
+import java.util.List;
+
+/**
+ * Lists all persons in the address book to the user in sorted order.
```
Comment (Creastery) [inline src/seedu/addressbook/commands/SortCommand.java:8]: What is a `sorted order`?

File: src/seedu/addressbook/commands/SortCommand.java [hunk 3]
```diff
@@ -1,4 +1,26 @@
...
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ":\n"
+            + "Displays all persons in the address book as a list with index numbers in a sorted order.\n\t"
+            + "Example: " + COMMAND_WORD;
+
+
+    @Override
+    public CommandResult execute() {
+        List<ReadOnlyPerson> allPersons = addressBook.getAllPersons().immutableListView();
+        return new CommandResult(getMessageForPersonListShownSummary(allPersons), allPersons);
```
Comment (Creastery) [inline src/seedu/addressbook/commands/SortCommand.java]: Perhaps indicate in the commit message that this function simply returns the list without actually returning a sorted list of `allPersons` yet!

File: src/seedu/addressbook/commands/SortCommand.java [hunk 4]
```diff
@@ -19,8 +19,9 @@
 
     @Override
     public CommandResult execute() {
+        addressBook.sort();
         List<ReadOnlyPerson> allPersons = addressBook.getAllPersons().immutableListView();
-        return new CommandResult(getMessageForPersonListShownSummary(allPersons), allPersons);
+        return new CommandResult(getMessageForPersonSortShownSummary(allPersons), allPersons);
```
Comment (Creastery) [inline src/seedu/addressbook/commands/SortCommand.java:24]: You could squash this commit with the previous two commits.
```
Messages: add sort command output message
``` 
...suggests that there's an output message that can be seen (but is not shown to the user).

```
Command: add method to call output message
```
...suggests that the newly added method is called to print the output message (but it doesn't).

File: src/seedu/addressbook/commands/SortCommand.java (hunk 2 above)
Comment (sanjukta99) [inline src/seedu/addressbook/commands/SortCommand.java:8]: You must mention *alphabetically sorted / lexicographically sorted* order

File: src/seedu/addressbook/data/AddressBook.java [hunk 5]
```diff
@@ -65,6 +65,8 @@ public void clear() {
         allPersons.clear();
     }
 
+    public void sort() { allPersons.sort(); }
+
```
Comment (sanjukta99) [inline src/seedu/addressbook/data/AddressBook.java:69]: Use K&R style brackets as mentioned in the Java Coding Standards on the module website

File: docs/UserGuide.adoc [hunk 6]
```diff
@@ -54,6 +54,11 @@ Examples:
 Shows a list of all persons in the address book. +
 Format: `list`
 
+== Listing all persons : `sort`
+
+Shows a list of all persons in the address book in sorted order. +
+Format: `sort`
+
```
Comment (sanjukta99) [inline docs/UserGuide.adoc:61]: Good that you have updated the User Guide

File: test/java/seedu/addressbook/logic/LogicTest.java [hunk 7]
```diff
@@ -104,6 +104,24 @@ public void execute_help() throws Exception {
...
+        // prepare address book state
+        helper.addToAddressBook(addressBook, false, true);
+
+        assertCommandBehavior("sort",
+                Command.getMessageForPersonSortShownSummary(expectedList),
+                expectedAB,
+                true,
+                expectedList);
+    }
+
```
Comment (sanjukta99) [inline test/java/seedu/addressbook/logic/LogicTest.java:124]: Good that you have updated testing. Try to write some more test cases in the future

**Review by sanjukta99 — COMMENTED** [review]
Overall a good effort. Please read the comments and close this PR.

**Pull request closed without merging**